packages/node_modules/pouchdb-core/src/changes.js

```diff
@@ -169,6 +170,11 @@ Changes.prototype.doChanges = function (opts) {
     opts.filter = '_view';
   }
 
+  /* istanbul ignore if */
+  if (opts.selector && !opts.filter) {
```

Comment:
On investigation, the CouchDB will only use the `selector` filter in the body if `filter=_selector` is specified in the query string. The implementation in this PR deviates from this (similar to our `doc_ids` filter), in that we do not require the `_selector` filter to be specified - we inject it automatically if a `selector` is provided. My inclination is to add validation that throws in the case that both `filter` and `selector` are specified, and `filter` is not `_selector`.